lua/include/packet.lua

```diff
@@ -182,132 +134,278 @@ function pkt:enableTimestamps()
+	
+	local packet = {}
+	packet.__index = packet
+	
+	packet.defaultArgs = defaultArgs
+
+	-- create struct and offer cast
+	local name, ctype = makeStruct(args)
+	packet.name = name
+
+	local get = function(self) return ctype(self.pkt.data) end
+
+	-- private/helper functions of the packet
+	packet.getHeaders = function(self) 
+		local headers = {} 
+		for i, v in ipairs(args) do 
+			headers[i] = _getHeader(self, v) 
+		end 
+		return headers 
+	end
 
-local arpPacketType = ffi.typeof("struct arp_packet*")
---- Retrieve an ARP packet.
--- @return Packet in 'struct arp_packet' format
-function pkt:getArpPacket()
-	return arpPacketType(self.pkt.data)
-end
+	-- public functions of the packet
+	packet.dump = function(self, bytes) _packetDump(self, bytes) end
+	
+	packet.fill = function(self, namedArgs) _packetFill(self, namedArgs) end
 
-local ptpPacketType = ffi.typeof("struct ptp_packet*")
---- Retrieve an PTP packet.
--- @return Packet in 'struct ptp_packet' format
-function pkt:getPtpPacket()
-	return ptpPacketType(self.pkt.data)
-end
+	packet.get = function(self) return _packetGet(self) end
 
-local ip4PacketType = ffi.typeof("struct ip_packet*")
---- Retrieve an IP4 packet.
--- @return Packet in 'struct ip_packet' format
-function pkt:getIP4Packet()
-	return ip4PacketType(self.pkt.data)
-end
+	packet.resolveLastHeader = function(self) return _packetResolveLastHeader(self) end
 
-local ip6PacketType = ffi.typeof("struct ip_v6_packet*")
---- Retrieve an IP6 packet.
--- @return Packet in 'struct ip_v6_packet' format
-function pkt:getIP6Packet()
-	return ip6PacketType(self.pkt.data)
-end
+	packet.setLength = function(self, length) _packetSetLength(self, length) end
 
---- Retrieve either an IPv4 or IPv6 packet.
--- @param ipv4 If true or nil returns IPv4, IPv6 otherwise
--- @return Packet in 'struct ip_packet' or 'struct ip_v6_packet' format
-function pkt:getIPPacket(ipv4)
-	ipv4 = ipv4 == nil or ipv4
-	if ipv4 then
-		return self:getIP4Packet()
-	else
-		return self:getIP6Packet()
+	-- functions for manual (not offloaded) checksum calculations
+	packet.calculateChecksums = function(self) _packetCalculateChecksums(self) end
+	
+	for _, v in ipairs(args) do
+		if v == 'icmp' then
+			packet.calculateIcmpChecksum = function(self) self.icmp:calculateChecksum() end
+		end
 	end
-end
 
-local icmp4PacketType = ffi.typeof("struct icmp_packet*")
---- Retrieve an ICMPv4 packet.
--- @return Packet in 'struct icmp_packet' format
-function pkt:getIcmp4Packet()
-	return icmp4PacketType(self.pkt.data)
+	ffi.metatype(name, packet)
+	return get
 end
 
-local icmp6PacketType = ffi.typeof("struct icmp_v6_packet*")
---- Retrieve an ICMPv6 packet.
--- @return Packet in 'struct icmp_v6_packet' format
-function pkt:getIcmp6Packet()
-	return icmp6PacketType(self.pkt.data)
-end
+--- Print a hex dump of a packet.
+-- @param self the packet
+-- @param bytes Number of bytes to dump. If no size is specified the payload is truncated.
+function _packetDump(self, bytes) 
```

Comment:
why are this, and the following, functions globals?

shouldn't it be just 
    local function packetDump(self, bytes)
?